src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -1129,7 +1129,9 @@ private boolean needIndexOrdering()
      */
     public SortedRowsBuilder sortedRowsBuilder(int limit, int offset, QueryOptions options)
     {
-        if (!needsPostQueryOrdering() || orderingComparator == null)
+        assert (orderingComparator != null) == needsPostQueryOrdering();
```

Comment:
This has one downside. If it fails, we don't know if it fails `true != false` or `false != true`.
Can you add a message?

Reply:
Added.